src/Models/BaseElement.php

```diff
@@ -319,16 +319,18 @@ public function write(
         return parent::write($showDebug, $forceInsert, $forceWrite, $writeComponents, $skipValidation);
     }
 
+    protected function onBeforeWrite()
+    {
+        parent::onBeforeWrite();
+        $this->ensureSortSet();
+    }
+
     /**
      * Increment the sort order if one hasn't been already defined. This
      * ensures that new elements are created at the end of the list by default.
-     *
-     * {@inheritDoc}
      */
-    public function onBeforeWrite()
+    public function ensureSortSet()
```

Comment:
This method has been added so that Resolver.php can call it directly, previously it was calling onBeforeWrite() directly